### DISCUSSION TITLE
個別の記事に編集者をクレジットするのは良いアイデアだと思うのですがいかがでしょう?

### DISCUSSION THREAD

**Discussion by kakutani** [rubima/magazine.rubyist.net#441] [要望・アイデア]
#431 からこっちにもってきます。

>るびまの方針を確認しないでのジャストアイデアですが、「編集：角谷信太郎」クレジットしませんか！
_Originally posted by @k16shikano in https://github.com/rubima/magazine.rubyist.net/issues/431#issuecomment-1581666144_

とか、これは共著にしませんか、という感じですが…:

> 他にもレビューやコメントをいただいた方はたくさんいますが、この記事は羽角さんと二人で書いたということにしたいんですが、いかがでしょう。もし嫌でなければ私としてはぜひ...!
_Originally posted by @neko314 in https://github.com/rubima/magazine.rubyist.net/issues/385#issuecomment-1574879110_

https://github.com/rubima/magazine.rubyist.net/pull/431#issuecomment-1581690018 にも書きましたが、私はPragmatic Bookshelfフォロワーなので、編集した人をクレジットするののは良いプラクティスだと思います! (目を通してくれた人たちもいるんだし、なんなら記事からPRのリンクがあってもよさそう)

WDYT? @takkanm 

補足:
Pragmatic Bookshelfの書目はある時期(Pragmatic Version Control with Gitぐらい)からカバーに編集者をクレジットしているのでした<details>
<summary>最新刊での例</summary>

![image](https://github.com/rubima/magazine.rubyist.net/assets/2160/2135ed79-8f40-479b-af83-ebbb0a292ac3)
</details>

**Comment by neko314** [comment]
わたしは大賛成です😊

**Reply by kakutani** [reply]
賛同者が我々(わたしと鹿野さん)以外にも陽にあらわれたので #446 をつくってみました。
(PRからDiscussionに対しては、Issueのようには参照のリンクがつくられないのだな…)

**Comment by takkanm** [comment]
TL;DR: 基本はいいと思うんですが、だれがいついれるのがクリアになってるとよさそう

description を読むと、課題が混ざっちゃっている気がします。私の理解として、今の description ででている課題としては、

1. 編集者をクレジットしたい
2. レビュワーしてくれた人にメンションしたい
3. neko314 さんの記事をはすみさんと共著という扱いにしたい

です。

私の意見を書く前にいったん私のコメントにおける用語の定義を再確認させてもらいます。

- 編集者 - 記事をるびまにだすための手助け(必要な情報を教えたり、校正したり、mergeしたり)をする人
- レビュワー - PR にコメントで指摘したりしてくれた人たち

まず、1 について。誰がどのタイミングで記事に書くか ( PR を作った人とるびま編集者が違うと編集が記事を変更するのはちょっとめんどいというのが前号の気づき)というのが気になるところはあるのですが、いいんじゃないかと思います。

2 について。これは PR でオープンに事前の記事をやりとりするようになってでてきた課題かと思います。個人的には、書籍でもあるようにレビュワーには謝辞をいれたいと感じています。前号をリリースしたときは[編集後記](https://magazine.rubyist.net/articles/0062/0062-EditorsNote.html)で謝辞を述べるようにしていました。なので、Special Thanks とかで記事最後とかに、最初のリリースまでにレビューにしてくれた人をクレジットするとよさそうかと思います。

3 については、neko314 さんと hasmikin さんの間で調整してもらえればおっけーです。ただ、私が見ていた範疇としては、共著というよりはあれは本物の編集の仕事だなぁと思いました。

**Comment by k16shikano** [comment]
FYIとして持論を置いときますね

- 原稿の執筆をいっしょにやったという意識があるのは共著
- 読者が読む状態にオーナーシップをもって関与したのは編集
- コメントをしたのは量によらずレビュー

**Comment by kakutani** [comment]
枝の活用がむずかしい(のでtop postしよう)。

> 3 については、neko314 さんと hasmikin さんの間で調整してもらえればおっけーです。ただ、私が見ていた範疇としては、共著というよりはあれは本物の編集の仕事だなぁと思いました。

@neko314 ↑の @k16shikano の「持論」を踏まえると、私も今回の @hasumikin の活躍は「編集」かなあと思うのですが、いかがでしょう。

> 2 について。これは PR でオープンに事前の記事をやりとりするようになってでてきた課題かと思います。個人的には、書籍でもあるようにレビュワーには謝辞をいれたいと感じています。前号をリリースしたときは[編集後記](https://magazine.rubyist.net/articles/0062/0062-EditorsNote.html)で謝辞を述べるようにしていました。なので、Special Thanks とかで記事最後とかに、最初のリリースまでにレビューにしてくれた人をクレジットするとよさそうかと思います。

鹿野持論のレビュワーの皆さんは編集後記でクレジットする、のが良さそう(行儀も良いと思います)

> まず、1 について。誰がどのタイミングで記事に書くか ( PR を作った人とるびま編集者が違うと編集が記事を変更するのはちょっとめんどいというのが前号の気づき)というのが気になるところはあるのですが、いいんじゃないかと思います。

個別の記事に載せるのは個別の対応でいいかな〜と思ってます。現状 #446 でも設定は任意だし、
きっかけについても「書いた人」の提案なので https://github.com/rubima/magazine.rubyist.net/pull/431#issuecomment-1581666144 。追加する場所もるびまヘッダなのでまあ「記事を変更する」といってもメタデータだし。
本文に「編集した人について」は不要そう。なぜなら編集後記があるから。

ところで0062の編集後記、せっかくいいことが書いてあるのに、段落分けされてないから怪文書感があって勿体無いですね…。
(もしリストになっているならheadingとtextで書くといいんじゃないだろうか)

**Reply by kakutani** [reply]
#446 に approve がほしい! @takkanm

**Reply by takkanm** [reply]
> 個別の記事に載せるのは個別の対応でいいかな〜と思ってます。現状 #446 でも設定は任意だし、
> きっかけについても「書いた人」の提案なので [#431 (comment)](https://github.com/rubima/magazine.rubyist.net/pull/431#issuecomment-1581666144) 。追加する場所もるびまヘッダなのでまあ「記事を変更する」といってもメタデータだし。

ここで「記事」とよんでいたのは、markdown のファイルをさしていました。記事を書いた人が fork して PR を出した場合、直接 PR をいじれないので、ファイルを編集がいじるには以下の方法になります。

- PR に PR or suggested change をだして merge してもらう
- fork したリポジトリに混ぜてもらう
- merge 後直接編集する

となると、誰がどのタイミングでというのがきになったのでした。

> 本文に「編集した人について」は不要そう。なぜなら編集後記があるから。

その理由だと悩ましい気がしました。なぜなら編集後記は慣例的には自己紹介ではないのなので。例えると、n 月間 LambdaNote の編集部から、WebDBPress の編集後記のようなものですね。なので、「人について」みたいな人となりがわかるような自己紹介もあっていい気がしますが、それが個別の記事に全てのるみたいな状況は避けたい気がします。(記事は記事が主役)

ちなみにその二つの雑誌については、個別の記事に編集の人の名前はクレジットされていないんですよね。
で、るびまは Web マガジンだというスタンスに戻ると、デリバラブルな単位としては記事だけどパッケージとしては号になるので、編集後記に編集したひとの名前がのっている状態でもいいのではという気もしてきましたがどうでしょう。

**Reply by kakutani** [reply]
> となると、誰がどのタイミングでというのがきになったのでした。

なるほどたしかに。

> それが個別の記事に全てのるみたいな状況は避けたい気がします。(記事は記事が主役)

記事が主役だからといって、編集した人の名前が個別に出なくていいとは思わないですが、その話を引っ張りたいわけではないので、その話はここまでで。実際のところ、1人がn記事編集することになりそうだからまとめて編集後記でいいんじゃない、となれば現行どおりっすね。

> ちなみにその二つの雑誌については、個別の記事に編集の人の名前はクレジットされていないんですよね。

かならずしも既存のものに倣う必要はないと思うし、n月刊ラムダノートは編集者の名前をもっとわかりやすく出したほうがいいと思う派ですが(それこそ表紙にPragProgみたいにeditid byって載せてもいいとおもう)、これもまた別の話だな。

今回は実運用上の懸念2点の解決案をスッとは思いつかないので、#446 は一旦引っ込めます。

* どのタイミングでforkからのPRの原稿に手を加えるかが悩ましい
* デリバリーは記事単位だとしてもパッケージは号単位だから編集後記でいいんじゃない?

あ。「著者→書いた人」はやりたいから、それだけ別PRにしようと思います。

**Reply by kakutani** [reply]
> あ。「著者→書いた人」はやりたいから、それだけ別PRにしようと思います。

おっと。昔の記事でも表紙は「書いた人」だけど記事本文は「著者について」が多めだから、これもそのままでよさそう。

**Comment by neko314** [comment]
なるほど編集💡しっくりきます。

**Comment by hasumikin** [comment]
@neko314 
ruby-jpから飛んできました。はい、編集を手伝いましたという体で名前を載せてもらうのでOKです。
あとは、、どう載せるか？ですよね？
たんに名前（hasumikin）だけで構わないですが、もし全体の体裁上、プロフィルあったほうが収まりがよいようでしたら、「PicoRubyとPRK Firmwareの作者」ということを加えていただけたら幸いです

**Reply by neko314** [reply]
@hasumikin
ありがとうございます！
プロフィール載せる方向でPRにしてみます。

**Reply by kakutani** [reply]
@neko314 振り回しちゃって恐縮です。 今回については、編集した人の記事での個別クレジットはやらないので(https://github.com/rubima/magazine.rubyist.net/discussions/441#discussioncomment-6197449)
* #385 の「この記事を書いた人の感想」のセクションあたりに、hasumikinへの謝辞を @neko314 が書く
または/かつ
* 0063の編集後記に @hasumikin も登場する

でおねがいしたいです!

**Reply by neko314** [reply]
ありがとうございます。
単純に項目増やして書けばいいって思っていたのですが、考えることがたくさんあったんですね〜
今回は了解です！謝辞は追記します😸

**Comment by neko314** [comment]
このアイディア良いなと思っているのですが、個別だと「この人は編集だがこの人は編集ではない」という線引きが難しい場合があるなと感じています。その号にコミットした人一覧のようなものを表紙の最後か編集後記あたりに載せられると良いかもしれないです。あるいは、それ用のページを新設する。
後からPRでtypo修正した場合でもPRをマージしたらそこに追加されるような仕組みがあると良いかもしれません。私だったら、そこに自分が載っていたら嬉しいと思う。
